src/plugins/dashboard/public/plugin.tsx

```diff
@@ -308,6 +316,47 @@ export class DashboardPlugin
       stopUrlTracker();
     };
 
+    const registerDashboardProvider: RegisterDashboardProviderFn = (
+      provider: DashboardProvider
+    ) => {
+      const found = this.dashboardProviders[provider.savedObjectsType];
+      if (found) {
+        throw new Error(`DashboardProvider ${provider.savedObjectsType} is registered twice`);
+      }
+      if (
+        isEmpty(provider.createSortText) ||
+        isEmpty(provider.createUrl) ||
+        isEmpty(provider.createLinkText)
+      ) {
+        throw new Error(
+          `DashboardProvider ${provider.savedObjectsType} requires 'createSortText', 'createLinkText', and 'createUrl'`
+        );
+      }
+      if (isEmpty(provider.savedObjectsType || isEmpty(provider.savedObjectsName))) {
+        throw new Error(
+          `DashboardProvider ${provider.savedObjectsType} requires 'savedObjectsId', and 'savedObjectsType'`
+        );
+      }
+
+      this.dashboardProviders[provider.savedObjectsType] = provider;
+    };
+
+    registerDashboardProvider({
```

Comment:
Dashboard always registers itself as a "Dashboard Provider".  This way, the usage of the providers list is unambiguous and does not require conditional processing.